addons/project/models/project.py

```diff
@@ -2003,8 +2004,8 @@ def create(self, vals_list):
                 vals['date_assign'] = fields.Datetime.now()
                 if not project_id:
                     user_ids = self._fields['user_ids'].convert_to_cache(vals.get('user_ids', []), self)
-                    if self.env.user.id not in user_ids:
-                        vals['user_ids'] = [Command.set(list(user_ids) + [self.env.user.id])]
+                    if not is_superuser:
+                        vals['user_ids'] = [Command.set(list(user_ids) + [self._uid])]
```

Comment:
Prefer using `self.env.uid`
```suggestion
                        vals['user_ids'] = [Command.set(list(user_ids) + [self.env.uid])]
```